app/vmagent/remotewrite/client.go

```diff
@@ -84,43 +86,51 @@ type client struct {
 	stopCh chan struct{}
 }
 
-func newClient(argIdx int, remoteWriteURL, sanitizedURL string, fq *persistentqueue.FastQueue, concurrency int) *client {
```

Comment:
How about just renaming this function to `newHTTPClient` and not modifying it contents except of setting `c.sendBlock = c.sendBlockHTTP`? Then the caller could determine which client to create from the remoteWriteURL scheme and call the corresponding function. E.g. if the scheme is `foobar`, then the caller could call `newFoobarClient()`?

Reply:
Sounds good to me!